api/pkg/apis/v1alpha1/providers/target/mock/mock.go

```diff
@@ -124,7 +124,7 @@ func (m *MockTargetProvider) Apply(ctx context.Context, deployment model.Deploym
 		for i, _ := range cache[m.Config.ID] {
 			if cache[m.Config.ID][i].Name == c.Component.Name {
 				found = true
-				if c.Action == "delete" {
+				if c.Action == model.ComponentDelete {
```

Comment:
This is inconsistent. Whether we do case-sensitive comparison everywhere, or we do case-insensitive comparison everywhere

Reply:
In my opinion, for internal state, we need to use enum values and use case sensitive comparison with enum values, while for inputs from users. we need to use case insensitive comparison for better user experience.